api/base/serializers.py

```diff
@@ -29,6 +29,65 @@ def _url_val(val, obj, serializer, **kwargs):
         return val
 
 
+class HyperlinkedIdentityFieldWithMeta(ser.HyperlinkedIdentityField):
+    # Returns a list with a url and an optional hash containing additional meta information
+
+    def __init__(self, view_name=None, **kwargs):
+        kwargs['read_only'] = True
+        kwargs['source'] = '*'
+        self.meta = kwargs.pop('meta', None)
+        super(ser.HyperlinkedIdentityField, self).__init__(view_name, **kwargs)
+
+    def get_url(self, obj, view_name, request, format):
```

Comment:
This is an almost exact copy of the parent class's (`HyperlinkedRelatedField`) implementation. Can this be refactored or removed (e.g. by ensuring that `getattr(obj, self.lookup_field)` does not return `None`?